multi-stream.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+var realPino = require('./pino')
+var noop = require('./noop')
+
+function pino (opts, stream) {
+  if (opts && (opts.writable || opts._writableState)) {
+    return realPino(null, opts)
+  }
+
+  if (opts.hasOwnProperty('stream') === true) {
+    return realPino(opts, opts.stream)
+  }
+
+  if (opts.hasOwnProperty('streams') === false) {
+    return realPino(opts, stream)
+  }
+
+  if (Array.isArray(opts.streams) === false) {
+    return realPino(opts, opts.streams)
+  }
+
+  var streams = opts.streams
+  var loggers = {}
+  for (var i = 0, j = streams.length; i < j; i += 1) {
+    var _opts = Object.create(opts)
+    var s = streams[i]
+    _opts.level = (s.level) ? s.level : 'info'
+    if (loggers[_opts.level]) {
+      loggers[_opts.level].push(realPino(_opts, s.stream))
+    } else {
+      loggers[_opts.level] = [realPino(_opts, s.stream)]
+    }
+  }
+
+  function MSPino (_loggers) {
+    this.loggers = _loggers
+  }
+  MSPino.prototype = Object.create(realPino.Pino.prototype)
+  MSPino.constructor = MSPino
+
+  function doLog (logger, level, params) {
+    logger[level].apply(logger, params)
+  }
```

Comment:
probably you are better off in generating a flamegraph with 0x and seeing if there is an unoptimized method around.